### PR TITLE
VAGOV-6290: Removing deprecated vamc system fields.

### PR DESCRIPTION
**Pull request by ethanteague**
## What this does
- Removes legacy fields from vamc system
- Depends on https://github.com/department-of-veterans-affairs/vets-website/pull/11338 first being merged and deployed to prod

### CODE REVIEW

**Comment by swirtSJW** [comment]
Looks like we have a conflict here.

File: config/sync/core.entity_form_display.node.health_care_region_page.default.yml [hunk 1]
```diff
@@ -573,14 +567,4 @@ content:
...
-  created: true
-  field_email_subscription: true
-  field_email_subscription_links: true
-  field_link_facility_emerg_list: true
-  field_link_facility_news_list: true
-  field_links: true
-  field_sign_up_for_emergency_emai: true
-  promote: true
-  sticky: true
-  uid: true
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.health_care_region_page.default.yml]: I think these as well as 'created' got caught in the crossfire.

Comment (ethanteague) [reply config/sync/core.entity_form_display.node.health_care_region_page.default.yml]: Yeah? I'll admit I'm confused - I had a conflict I resolved where codebase didn't know if these fields should be hidden or visible, so I defaulted to visible.....if you say they should be visible, I defer to you, @swirtSJW - can you confirm?

Comment (swirtSJW) [reply config/sync/core.entity_form_display.node.health_care_region_page.default.yml]: Yah I think Kev wants those to remain hidden.

Comment (ethanteague) [reply config/sync/core.entity_form_display.node.health_care_region_page.default.yml]: Cool - thanks for catching that, @swirtSJW ! Just updated.

File: config/sync/core.entity_form_display.node.health_care_region_page.default.yml [hunk 2]
```diff
@@ -567,4 +567,14 @@ content:
     region: content
     settings: {  }
     third_party_settings: {  }
-hidden: {  }
+hidden:
+  created: true
+  field_email_subscription: true
+  field_email_subscription_links: true
+  field_link_facility_emerg_list: true
+  field_link_facility_news_list: true
```
Comment (swirtSJW) [inline config/sync/core.entity_form_display.node.health_care_region_page.default.yml]: Aren't these supposed to be going away?  I think ony created, promote, sticky, and promote and uid should remain.

Comment (ethanteague) [reply config/sync/core.entity_form_display.node.health_care_region_page.default.yml]: Yep, @swirtSJW - updated

File: config/sync/core.entity_form_display.node.health_care_region_page.default.yml (hunk 1 above)
Comment (kevwalsh) [reply config/sync/core.entity_form_display.node.health_care_region_page.default.yml]: yep, hidden. i made this change recently.

**Pull request merged**